projects/storefrontlib/src/cms-components/checkout/components/po-number/po-number.component.ts

```diff
@@ -23,41 +24,95 @@ export class PoNumberComponent {
   readonly ACCOUNT_PAYMENT = 'ACCOUNT';
 
   @ViewChild('poNumber', { static: false })
-  poNumberInput: ElementRef;
+  private _poNumberInput: ElementRef;
 
-  typeCode: string;
-  typeSelected$: Observable<
-    string
-  > = this.paymentTypeService.getSelectedPaymentType().pipe(
-    filter((selected) => selected !== undefined),
-    tap((selected) => (this.typeCode = selected))
-  );
-
-  poNumber: string;
-  poNumber$: Observable<string> = this.paymentTypeService.getPoNumber().pipe(
-    filter((po) => !!po),
-    tap((po) => (this.poNumber = po))
-  );
+  private _costCenterId: string;
+  set costCenterId(code: string) {
+    this._costCenterId = code;
+  }
+  get costCenterId(): string {
+    return this._costCenterId;
+  }
 
-  costCenters$: Observable<
-    CostCenter[]
-  > = this.costCenterService.getActiveCostCenters();
+  paymentTypeCode: string;
+  cartPoNumber: string;
+  cartCostCenterId: string;
 
-  backBtnText = this.checkoutStepService.getBackBntText(this.activatedRoute);
+  private allowRedirect = true;
 
   constructor(
     protected paymentTypeService: PaymentTypeService,
     protected checkoutStepService: CheckoutStepService,
-    protected costCenterService: UserCostCenterService,
+    protected userCostCenterService: UserCostCenterService,
+    protected checkoutCostCenterService: CheckoutCostCenterService,
     protected activatedRoute: ActivatedRoute
   ) {}
 
+  get typeSelected$(): Observable<string> {
+    return this.paymentTypeService.getSelectedPaymentType().pipe(
+      filter((selected) => selected !== undefined),
+      tap((selected) => (this.paymentTypeCode = selected))
+    );
+  }
+
+  get cartPoNumber$(): Observable<string> {
+    return this.paymentTypeService.getPoNumber().pipe(
+      filter((po) => po !== undefined),
+      tap((po) => (this.cartPoNumber = po))
+    );
+  }
+
+  get costCenters$(): Observable<CostCenter[]> {
+    return this.userCostCenterService.getActiveCostCenters().pipe(
+      filter((costCenters) => costCenters && costCenters.length > 0),
+      tap((costCenters) => {
+        if (this._costCenterId === undefined) {
+          this._costCenterId = costCenters[0].code;
+        }
+      })
+    );
+  }
+
+  get cartCostCenter$(): Observable<string> {
+    return this.checkoutCostCenterService.getCostCenter().pipe(
+      filter((cc) => !!cc),
```

Comment:
There is some convention it seems to use Boolean() instead of !!

In this specific case I think it could even become `.filter(Boolean)`

Reply:
done, but `.filter(Boolean)` has compile error, has to use `filter((cc) => Boolean(cc))`.